lib/discourse_hub.rb

```diff
@@ -4,18 +4,27 @@
 module DiscourseHub
 
   def self.version_check_payload
-    {
-      installed_version: Discourse::VERSION::STRING
-    }.merge!( Discourse.git_branch == "unknown" ? {} : {branch: Discourse.git_branch})
+    get_payload.merge!( Discourse.git_branch == "unknown" ? {} : {branch: Discourse.git_branch})
   end
 
   def self.discourse_version_check
     get('/version_check', version_check_payload)
   end
 
+  def self.stats_fetched_at=(time_with_zone)
+    $redis.set stats_fetched_at_key, time_with_zone.as_json
+  end
 
   private
 
+  def self.get_payload
+    if SiteSetting.send_anonymized_statistics_to_discourse && stats_fetched_at < 7.days.ago
+      About.fetch_cached_stats.symbolize_keys.merge!({ installed_version: Discourse::VERSION::STRING })
+    else
+      { installed_version: Discourse::VERSION::STRING }
```

Comment:
We can probably move `{ installed_version: Discourse::VERSION::STRING }` into `.version_check_payload` as `default_payload`.